src/SHA1.elm

```diff
@@ -373,24 +374,84 @@ So in the recursion, `b16` is dropped, all the others shift one position to the
 Then the `deltaState` is also updated with the `value`.
 
 -}
-reduceWords i deltaState b16 b15 b14 b13 b12 b11 b10 b9 b8 b7 b6 b5 b4 b3 b2 b1 =
-    if (i - blockSize) < 0 then
+reduceWords i ((DeltaState { a, b, c, d, e }) as deltaState) b16 b15 b14 b13 b12 b11 b10 b9 b8 b7 b6 b5 b4 b3 b2 b1 =
```

Comment:
This function now performs 8 calculations per iteration.That means fewer function calls, but diminishing returns mean that it's not worth it to go up to 16.